Makefile

```diff
@@ -45,17 +45,12 @@ clusterctl: fmt vet
 	go build -o bin/clusterctl ./cmd/clusterctl
 
 clusterctl-in-docker:
-	docker run -v $(CWD):/go/src/sigs.k8s.io/cluster-api-provider-vsphere \
+	docker run --rm -v $(CWD):/go/src/sigs.k8s.io/cluster-api-provider-vsphere \
```

Comment:
Can you add 
```
-u $$(id -u):$$(id -g) \
```
here, I'm seeing this:
```
$ make prod-yaml
rm: cannot remove '/home/ubuntu/go/src/sigs.k8s.io/cluster-api-provider-vsphere/.gopath/src/sigs.k8s.io/cluster-api-provider-vsphere': Permission denied
hack/shell-with-gopath.sh: 29: cd: can't cd to /home/ubuntu/go/src/sigs.k8s.io/cluster-api-provider-vsphere/.gopath/src/sigs.k8s.io/cluster-api-provider-vsphere
CAPV_MANAGER_IMAGE=andrewsykim/cluster-api-provider-vsphere:net hack/generate-yaml.sh
rm: cannot remove '/home/ubuntu/go/src/sigs.k8s.io/cluster-api-provider-vsphere/.gopath/src/sigs.k8s.io/cluster-api-provider-vsphere': Permission denied
hack/shell-with-gopath.sh: 29: cd: can't cd to /home/ubuntu/go/src/sigs.k8s.io/cluster-api-provider-vsphere/.gopath/src/sigs.k8s.io/cluster-api-provider-vsphere
Makefile:126: recipe for target 'prod-yaml' failed
make: *** [prod-yaml] Error 2
```
and I suspect it's related to the user running in the docker container here

Reply:
Hi @andrewsykim,

I'm aware of this type of issue. It's because symlinks in a container when the user is running as root don't translate nicely to the host. However, this should likely fail the same way on `master`. Why not file a fix in a different PR?

Reply:
Hi @andrewsykim,

Ohhh, never mind. I thought this was for a different target due to the target listed above your comment.

Reply:
Fixing.

Reply:
Wait.. I'm going to ping you. I'm not sure where you want this added.

Reply:
Setting a user causes this isse though 
```
make clusterctl-in-docker
docker run --rm -v /home/ubuntu/go/src/sigs.k8s.io/cluster-api-provider-vsphere:/go/src/sigs.k8s.io/cluster-api-provider-vsphere \
  -w /go/src/sigs.k8s.io/cluster-api-provider-vsphere \
  -u $(id -u):$(id -g) \
  -e GOOS -e GOHOSTOS golang:1.12 \
  make clusterctl
hack/update-generated.sh codegen
update-generated: deepcopy pkg=./pkg/apis/vsphereproviderconfig/v1alpha1
failed to initialize build cache at /.cache/go-build: mkdir /.cache: permission denied
Makefile:86: recipe for target 'generate' failed
make: *** [generate] Error 1
Makefile:48: recipe for target 'clusterctl-in-docker' failed
make: *** [clusterctl-in-docker] Error 2
```